algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -38,23 +38,21 @@ package com.twitter.algebird
  */
 
 object QTree {
-  def apply[A: Monoid](kv: (Double, A), level: Int = -16): QTree[A] = {
+  def apply[A](kv: (Double, A), level: Int = -16): QTree[A] =
     QTree(math.floor(kv._1 / math.pow(2.0, level)).toLong,
       level,
       1,
       kv._2,
       None,
       None)
-  }
 
-  def apply[A: Monoid](kv: (Long, A)): QTree[A] = {
+  def apply[A](kv: (Long, A)): QTree[A] =
     QTree(kv._1,
       0,
       1,
       kv._2,
       None,
       None)
-  }
 
   /**
    * The common case of wanting a count and sum for the same value
```

Comment:
It's badly expressed. The point is that for a given bin with bounds (x,y), you are not just counting all k_i where x <= k_i < y, but also summing k_i. To put it another way, you are computing not just the curve of the distribution but the area under that curve (more accurately than if you did some kind of trapezoidal estimation).

Honestly this is just there because sometimes you only really care about the counts, and this is as useful a default for the sum as any.